lib/services/platform-service.ts

```diff
@@ -217,6 +217,16 @@ export class PlatformService implements IPlatformService {
 			this.ensurePlatformInstalled(platform).wait();
 			let changesInfo = this.$projectChangesService.checkForChanges(platform);
 			if (changesInfo.hasChanges) {
+				// android build artifacts need to be cleaned up when switching from release to debug builds
+				if (platform.toLowerCase() === "android") {
+					let previousPrepareInfo = this.$projectChangesService.getPrepareInfo(platform);
+					// clean up prepared plugins when not building for release
+					if (previousPrepareInfo && previousPrepareInfo.release && !this.$options.release) {
```

Comment:
Isn't it good idea to call the clean method in case the previous build was for debug config and the current one is for release? For example:
```TypeScript
if (previousPrepareInfo && previousPrepareInfo.release !== this.$options.release) {
```